vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp

```diff
@@ -145,6 +148,94 @@ void NativeAnimatedNodeManager::StopAnimation(int64_t animationId) {
   }
 }
 
+void NativeAnimatedNodeManager::RestartTrackingAnimatedNode(
+    int64_t animationId,
+    int64_t animatedToValueTag,
+    const std::shared_ptr<NativeAnimatedNodeManager> &manager) {
+  if (m_activeAnimations.count(animationId)) {
+    if (const auto animation = m_activeAnimations.at(animationId).get()) {
+      auto const animatedValueTag = animation->AnimatedValueTag();
+      auto const animationConfig = animation->AnimationConfig();
+      auto const endCallback = animation->EndCallback();
+      animation->StopAnimation(true);
+      m_activeAnimations.erase(animationId);
+      StartTrackingAnimatedNode(
+          animationId,
+          animatedValueTag,
+          animatedToValueTag,
+          animationConfig,
+          endCallback,
+          manager,
+          false);
+    }
+  }
+}
+
+void NativeAnimatedNodeManager::StartTrackingAnimatedNode(
+    int64_t animationId,
+    int64_t animatedNodeTag,
+    int64_t animatedToValueTag,
+    const folly::dynamic &animationConfig,
+    const Callback &endCallback,
+    const std::shared_ptr<NativeAnimatedNodeManager> &manager,
+    bool track) {
+  auto updatedAnimationConfig = animationConfig;
+  for (auto iterator = m_activeAnimations.begin();
+       iterator != m_activeAnimations.end();
+       iterator++) {
+    if (iterator->second->AnimatedValueTag() == animatedToValueTag) {
+      updatedAnimationConfig.insert(
+          static_cast<folly::StringPiece>(s_toValueIdName),
+          iterator->second->ToValue());
+
+      switch (AnimationTypeFromString(
+          animationConfig.find("type").dereference().second.getString())) {
+        case AnimationType::Frames:
+          updatedAnimationConfig.insert(
+              static_cast<folly::StringPiece>(s_framesName),
+              [animationConfig, iterator]() {
+                auto frames = folly::dynamic::array();
+                for (auto const &frame :
+                     animationConfig.find("frames").dereference().second) {
+                  frames.push_back(0.0);
+                }
+                for (auto const &frame : iterator->second->Frames()) {
```

Comment:
To append you can do `frames.insert(frames.end(), other.begin(), other.end())` (where other is iterator->second)

Reply:
Folly::Dynamic::array does not have an iterator based insert function.